bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/LinkBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.commons.link;
+
+import java.util.Map;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Builds a link and sets link properties.
+ */
+public interface LinkBuilder {
+
+    /**
+     * Sets the property name to read the link URL from.
+     *
+     * @param name The property name
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder setLinkURLPropertyName(@Nullable String name);
+
+    /**
+     * Sets the link target.
+     *
+     * @param target The link target
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder setLinkTarget(@Nullable String target);
+
+    /**
+     * Sets the link attributes.
+     *
+     * @param attributes The link attributes
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder setLinkAttributes(@NotNull Map<String, String> attributes);
```

Comment:
This method provides a flexible way to add HTML attributes to the link, e.g. `aria-label` or `title`.

Reply:
ok - still wondering if it might make sense to create dedicated methods like setLinkTarget for use cases like accessibility title which may affect lot of components.

Reply:
As there might be requirements to add specific attributes to a link, I would keep a flexible way to add them. We could replace it with a method to only add one attribute at a time, e.g.:
```
LinkBuilder setLinkAttribute(String key, String value);
```
In a first step I wouldn't add specific methods other than `setLinkTarget`. We could add specific methods later if needed.

Reply:
ok - yes, i like the setLinkAttribute approach better

Reply:
I updated the method in https://github.com/adobe/aem-core-wcm-components/pull/2196/commits/9aef6a37a154aff71b0a68608de42a54dc83b1d7